src/Morphir/JsonSchema/Backend.elm

```diff
@@ -105,55 +105,144 @@ extractTypes modName definition =
             )
 
 
-mapTypeDefinition : TypeName -> Type.Definition ta -> Maybe ( TypeName, SchemaType )
-mapTypeDefinition typeName definition =
+mapTypeDefinition : ( Path, Name ) -> Type.Definition ta -> List ( TypeName, SchemaType )
+mapTypeDefinition (( path, name ) as qualifiedName) definition =
     case definition of
-        Type.TypeAliasDefinition typeArgs typ ->
-            mapType typ
-                |> Maybe.map (Tuple.pair typeName)
-
-        Type.CustomTypeDefinition typeArgs accessControlledConstructors ->
-            Nothing
-
-
-mapType : Type ta -> Maybe SchemaType
+        Type.TypeAliasDefinition _ typ ->
+            [ ( mapQualifiedName qualifiedName
+              , case mapType typ of
+                    Ok val ->
+                        val
+
+                    _ ->
+                        Null
+              )
+            ]
+
+        Type.CustomTypeDefinition _ accessControlledCtors ->
+            let
+                oneOfs2 =
+                    accessControlledCtors.value
+                        |> Dict.toList
+                        |> List.map
+                            (\( ctorName, ctorArgs ) ->
+                                let
+                                    ctorNameString =
+                                        ctorName |> Name.toTitleCase
+                                in
+                                if List.isEmpty ctorArgs then
+                                    Const ctorNameString
+
+                                else
+                                    Array
+                                        (TupleType
+                                            (Const (ctorName |> Name.toTitleCase)
+                                                :: (-- begin
+                                                    ctorArgs
+                                                        |> List.map
+                                                            (\x ->
+                                                                case mapType (Tuple.second x) of
+                                                                    Ok val ->
+                                                                        val
+
+                                                                    _ ->
+                                                                        Null
```

Comment:
Same here. The right thing to do would be to propagate the error.

Reply:
Fixed now